### PR TITLE
Social: Add share status modal tracking

### PR DESCRIPTION
**Pull request by gmjuhasz**
Fixes https://github.com/Automattic/jetpack-reach/issues/548

## Proposed changes:

* Added tracking for the modal opening e34427a52ef6eab23153017a8467fd66affbfcdc
* Added tracking for the view and retry action e980de0cda6e1540efddf5cfdf6b95c88684655e

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/jetpack-reach/issues/548
## Does this pull request change what data or activity we track or use?

No.
## Testing instructions:

* Go to the post editor with this change
* Open browser console and run `localStorage.setItem('debug', 'dops:analytics');`

##### Opening the modal
* Create a new post and share it for the first time
* When the polling finishes, click the modal trigger to open the modal
* In the console you should see the click being tracked, and location be `post-publish-panel`
* Do the same from the editor on a shared post, the location should be `editor`

#### View action
* Open up the modal as before
* Click on View on a successfully shared share in the log.
* Inspect the console, you should see it being logged with the location of modal. That's added preemptively so when we do Phase 2 we can record the same event with a different prop

#### Retry action
* Open up the modal as before
* For a failed share, click retry.
* Inspect the console, you should see the action logged with the location, and the service prop. This is useful as we can monitor which service fails the most / getting retried the most.

#### Check events on Tracks.
* Wait ~10 minutes.
* Goto MC Live tracking
* Check the events there, and check the props. It should reflect what you saw in the logs before

Events to check:
* `jetpack_social_share_status_modal_opened`
* `jetpack_social_share_status_view`
* `jetpack_social_share_status_retry`

### CODE REVIEW

File: projects/js-packages/publicize-components/src/components/share-status/share-status-action.tsx [hunk 1]
```diff
@@ -39,12 +49,20 @@ export function ShareStatusAction( { connectionId, status, shareLink }: ShareSta
 		}
 
 		doPublicize( skippedConnections.map( connection => connection.connection_id ) );
-	}, [ connectionId, connections, doPublicize ] );
+	}, [ connectionId, connections, doPublicize, recordEvent, service ] );
+
+	const recordViewEvent = useCallback( () => {
+		recordEvent( 'jetpack_social_share_status_view', { location: 'modal' } );
```
Comment (manzoorwanijk) [inline projects/js-packages/publicize-components/src/components/share-status/share-status-action.tsx]: Should we pass `service` here as well?

**Review by manzoorwanijk — DISMISSED** [review]
Works fine. Just have an optional suggestion.

File: projects/js-packages/publicize-components/src/components/share-status/share-status-action.tsx (hunk 1 above)
Comment (gmjuhasz) [reply projects/js-packages/publicize-components/src/components/share-status/share-status-action.tsx]: Done!

**Review by manzoorwanijk — DISMISSED** [review]
Thank you for making the change

**Review by manzoorwanijk — APPROVED** [review]
This looks good now. Thanks

**Pull request merged**